### PR TITLE
Move setup properties out of setup.py in to setup.cfg

### PR DESCRIPTION
**Pull request by ashb**
I've moved all the ones that are "static" -- any form of dynamic or interpolated values are left in setup.py

If a value is passed as n kwrg to setup and in setup.cfg, the kwarg wins out.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by ashb** [comment]
I'm just moving across install_reuirements too

**Comment by potiuk** [comment]
I had a quick look before going to bed - and there is nothing that I would not like in this change @ashb :). As long as all the cases we had (installing locally, installing via Github llnk, installing via PyPI) work - this is all fine. 

There are a few (or maybe just one - INSTALL_REQUIREMENTS order - check_install_and_setup_requires in `pre_commit_check_order_setup.py` ) static check/pre-commit that you will find that need updating. I think that's all - but we should also carefully look at the artifacts generated by the PR to be sure of that (especially whether the docker images built contain all providers and whether the provider packages have the right dependencies etc. 

Nothing controversial in this PR really :)

**Comment by ashb** [comment]
Yeah the possible contriversal stuff (version detection, extra requires) needed more brain power to think of that I have tonight. So I'm doing sort of mechanical easy-to-do changes :D

**Comment by ashb** [comment]
Updated the sort checking in pre-commit. Setup tool provides a `read_configuration` function we can use.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -779,7 +779,8 @@ jobs:
           cache-name: cache-kubernetes-tests-v7-master
         with:
           path: ".build/.kubernetes_venv*"
-          key: "venv-${{ env.cache-name }}-${{ github.job }}-${{ hashFiles('setup.py') }}\
+          key: "venv-${{ env.cache-name }}-${{ github.job }}-${{ hashFiles('setup.py')\
+-${{ hashFiles('setup.cfg') }}\
```
Comment (potiuk) [inline .github/workflows/ci.yml:783]: Good point!

File: docs/installation.rst [hunk 2]
```diff
@@ -27,7 +27,7 @@ Getting Airflow
 Airflow is published as ``apache-airflow`` package in PyPI. Installing it however might be sometimes tricky
 because Airflow is a bit of both a library and application. Libraries usually keep their dependencies open and
 applications usually pin them, but we should do neither and both at the same time. We decided to keep
-our dependencies as open as possible (in ``setup.py``) so users can install different version of libraries
+our dependencies as open as possible (in ``setup.cfg``) so users can install different version of libraries
```
Comment (potiuk) [inline docs/installation.rst]: Should not that be (for now at least) both setup.py and setup.cfg?

File: setup.cfg [hunk 3]
```diff
@@ -44,6 +46,111 @@ license_files =
...
+    thrift>=0.9.2
+    typing;python_version<"3.6"
+    typing-extensions>=3.7.4;python_version<"3.8"
+    tzlocal>=1.4,<2.0.0
+    unicodecsv>=0.14.1
+    werkzeug~=1.0, >=1.0.1
+
+[options.packages.find]
+exclude =
+    airflow.providers
```
Comment (potiuk) [inline setup.cfg]: I believe find: uses "find_packages" not the namespace ones (there is a separate find_namespace:)? So we probably do not need those "excludes" here.

**Review by potiuk — APPROVED** [review]
Love it! Few small Nits/

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -788,6 +789,7 @@ jobs:
         with:
           path: ".build/bin"
           key: "bin-${{ env.cache-name }}-${{ github.job }}-${{ hashFiles('setup.py') }}\
+-${{ hashFiles('setup.cfg') }}\
```
Comment (potiuk) [inline .github/workflows/ci.yml]: Actually here we can remove both setup.py and setup.cfg. The ./buld/bin content only depends on the version of tools used (helm//kind/kubectl) and it does not depend on setup.py nor setup.cfg in any way. That was probably a copy&paste problem.

Comment (ashb) [reply .github/workflows/ci.yml]: Will do. (I didn't check the logic here, just saw setup.py being fingerprinted and added setup.cfg)

File: setup.cfg (hunk 3 above)
Comment (ashb) [reply setup.cfg]: I just did a lift-and-shift of the `find_packages()` call from setup.py. You are right though.

**Pull request merged**